app/controllers/public_controller.rb

```diff
@@ -103,6 +103,22 @@ def contact_us
     @title = t('public.contact_us.title')
   end
 
+  def commission_histories
+    if params[:validator_id]
+      @validator = Validator.find(params[:validator_id])
+      @commission_histories = CommissionHistory.where(network: params[:network], validator_id: @validator.id)
+                                               .order(created_at: :desc)
```

Comment:
you can remove `order`, `page` and `per` from `if-else` and apply it one time after the condition.